File: python/8kyu/count_sheeps.py
import codewars_test as test


def count_sheeps(sheep: list) -> int:
    return sum(x for x in sheep if x)


# def count_sheeps(arrayOfSheeps):
#   return arrayOfSheeps.count(True)


array1 = [True,  True,  True,  False,
          True,  True,  True,  True,
          True,  False, True,  False,
          True,  False, False, True,
          True,  True,  True,  True,
          False, False, True,  True]

test.assert_equals(result := count_sheeps(array1), 17,
                   "There are 17 sheeps in total, not %s" % result)
G

File: typescript/8kyu/count_sheeps/count_sheeps.ts
export function countSheeps(arrayOfSheep: (boolean | undefined | null)[]): number {
    return arrayOfSheep.filter((x) => x).length
}

import { expect } from "chai";

var array1 = [true, true, true, false,
    true, true, true, true,
    true, false, true, false,
    true, false, false, true,
    true, true, true, true,
    false, false, true, true];

describe('countSheeps', function () {
    it("should work correctly", () => {
        const result1 = countSheeps(array1);
        expect(result1).to.equal(17, "There are 17 sheeps in total, not " + result1);
    });
});